### PR TITLE
Teams

### PR DESCRIPTION
**Pull request by herrmannplatz**

### CODE REVIEW

File: app/script/conversation/ConversationRepository.js [hunk 1]
```diff
@@ -103,7 +105,10 @@ z.conversation.ConversationRepository = class ConversationRepository {
       const cleared = [];
       const unarchived = [];
 
-      this.sorted_conversations().forEach(function(conversation_et) {
+      this.sorted_conversations().forEach((conversation_et) => {
+        if (this.active_team() != null && !z.util.StringUtil.includes(conversation_et.display_name(), this.active_team().id)) { // TODO check against id
```
Comment (lipis) [inline app/script/conversation/ConversationRepository.js]: maybe using local variables to make it easier to read?

```js
const hasActiveTeam = this.active_team() != null;
const .... = !z.util.StringUtil.includes(conversation_et.display_name(), this.active_team().id);
```

File: app/style/list/teams.less [hunk 2]
```diff
@@ -0,0 +1,37 @@
+.teams {
+  width: @teams-width;
+  position: absolute;
+  top: 0;
+  left: 0;
+  bottom: 0;
+  background: linear-gradient(to bottom, fade(#000,0%), fade(#000,64%));
```
Comment (lipis) [inline app/style/list/teams.less]: Spaces inside `fade` :)

```less
fade(#000, 0%), fade(#000, 64%)
```

File: app/script/conversation/ConversationRepository.js (hunk 1 above)
Comment (herrmannplatz) [reply app/script/conversation/ConversationRepository.js]: outdated

File: test/unit_tests/conversation/ConversationMapperSpec.js [hunk 3]
```diff
@@ -28,7 +28,7 @@ describe('Conversation Mapper', function() {
     conversation_mapper = new z.conversation.ConversationMapper();
   });
 
-  describe('map_conversation', () => {
+  describe('map_conversation', function() {
```
Comment (lipis) [inline test/unit_tests/conversation/ConversationMapperSpec.js]: but but but `() =>` 🗡

File: app/script/conversation/ConversationRepository.js [hunk 4]
```diff
@@ -568,17 +582,28 @@ z.conversation.ConversationRepository = class ConversationRepository {
...
+   * @param {string} [team_id] - Team ID in which the conversation should be searched
    * @returns {Promise} Resolves with the conversation with requested user
    */
-  get_one_to_one_conversation(user_et) {
+  get_one_to_one_conversation(user_et, team_id = this.active_team().id) {
     for (const conversation_et of this.conversations()) {
-      if ([z.conversation.ConversationType.ONE2ONE, z.conversation.ConversationType.CONNECT].includes(conversation_et.type())) {
+      if (team_id) {
+        if (conversation_et.type() === z.conversation.ConversationType.REGULAR) {
+          if (user_et.id === conversation_et.participating_user_ids()[0] && team_id === conversation_et.team_id) {
```
Comment (lipis) [inline app/script/conversation/ConversationRepository.js]: Put all those 3 in constants and compare them instead.. (with proper names of course)

```js
const xx = conversation_et.type() === z.conversation.ConversationType.REGULAR;
const yy = user_et.id === conversation_et.participating_user_ids()[0];
const zz = team_id === conversation_et.team_id;

```

File: app/script/conversation/ConversationRepository.js [hunk 5]
```diff
@@ -568,17 +582,28 @@ z.conversation.ConversationRepository = class ConversationRepository {
...
+            return Promise.resolve(conversation_et);
+          }
+        }
+      } else if ([z.conversation.ConversationType.ONE2ONE, z.conversation.ConversationType.CONNECT].includes(conversation_et.type())) {
         if (user_et.id === conversation_et.participating_user_ids()[0]) {
```
Comment (lipis) [inline app/script/conversation/ConversationRepository.js]: same here to avoid the extra sole `if`

Comment (lipis) [reply app/script/conversation/ConversationRepository.js]: `user_et.id === conversation_et.participating_user_ids()[0]` appears twice so it actually makes more sense..!

File: app/script/search/SearchRepository.js [hunk 6]
```diff
@@ -51,16 +51,27 @@ z.search.SearchRepository = class SearchRepository {
...
+        }
+
+        return user_ets
+          .sort((user_a, user_b) => {
+            if (is_username) {
+              return z.util.StringUtil.sort_by_priority(user_a.username(), user_b.username(), name);
+            }
+            return z.util.StringUtil.sort_by_priority(user_a.name(), user_b.name(), name);
+          })
+          .slice(0, 10);
```
Comment (lipis) [inline app/script/search/SearchRepository.js]: Magic number! Please use constant instead of `10`

File: app/script/team/TeamMapper.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+
+    if (data.id) {
+      team_et.id = data.id;
+    }
+
+    if (data.name) {
+      team_et.name(data.name);
+    }
+
+    return team_et;
```
Comment (lipis) [inline app/script/team/TeamMapper.js:63]: Can we use [`Object.assign()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) here instead?

File: app/style/common/variables.less [hunk 8]
```diff
@@ -113,16 +101,18 @@
 // Framework
 // ----------------------------------------------------------------------------
 @app-bg:              #fafafa;
+
+@teams-width:          64px;
```
Comment (lipis) [inline app/style/common/variables.less]: seems off by one space :D

File: app/script/team/TeamMapper.js (hunk 7 above)
Comment (gregor) [reply app/script/team/TeamMapper.js:63]: We could but it would kind of defeat the purpose of a mapper and not handle the observables.

**Comment by ffflorian** [comment]
:tada:

**Pull request merged**